### DISCUSSION TITLE
Cause of 422 response

### DISCUSSION THREAD

**Discussion by fenhl** [rwf2/Rocket#2879] [Questions] [answered]
I'm seeing 422 Unprocessable Content being returned from my web app. I don't have anything in my code that generates that status, and I can't find anything in the Rocket docs about it either. Is there any way to find out why this status is being returned, e.g. some additional info the `#[rocket::catch(default)]` handler has access to?

**Comment by szabgab** [comment]
As far as I know this is returned when one of the guards fail. Could you share a code snippet?

**Reply by szabgab** [reply]
To further expand on it, for example you have a GET or a POST request that expects a field, but that field is missing in the request. You can see an example of a POST request with a test where it returns `Status::UnprocessableEntity` (422) [here](https://rust.code-maven.com/slides/rust/rocket-echo-using-post.html)

**Reply by fenhl** [reply, marked as answer]
Here's the endpoint that's failing: https://github.com/midoshouse/midos.house/blob/dee453cf71600a6ce74e4aeb50b7ccf383e40d12/src/user.rs#L293-L294

The errors are happening on the path `/user/jimbo`, so it's someone misunderstanding how profile pages work (my user IDs are numerical), but according to the [`FromParam`](https://docs.rs/rocket/latest/rocket/request/trait.FromParam.html) docs, this should return 404, not 422:

> If the conversion fails, the incoming request will be forwarded to the next matching route, if any.

> Since there are no additional matching routes, this example will result in a 404 error for requests with invalid `id` values.

So maybe this is a bug in Rocket?

**Reply by the10thWiz** [reply]
As noted in the issue you opened, this is actually a bug in the documentation. The docs should be updated to say that it forwards with a status of `422`.

**Comment by raif-s-naffah** [comment]
hi @fenhl ,

from experience Rocket returns 422 when it's unable to match an incoming request to a known handler.

may be having a closer look at the logs (near the start where Rocket lists all configured routes, ctachers, etc...) can shed more light on why your handler is not being picked.

also suggest you have a closer look at the [grammar][1] of what is taken into account when defining a _route_.  as stated there, a route is determined by more than just an HTTP verb and path.

finally, if you still cannot see the problem, may be catching the error (see [catchers][2] in the guide) and doing some logging can give more info.


[1]: https://api.rocket.rs/v0.5/rocket/attr.route#grammar
[2]: https://rocket.rs/guide/v0.5/requests/#error-catchers
